packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -248,21 +236,51 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
   }
 
   // ====== RestAPI ======
-  RestOperation _restFunctionHelper(
-      {required String methodName, required RestOptions restOptions}) {
-    // Send Request cancelToken to Native
-    String cancelToken = UUID.getUUID();
 
-    Future<RestResponse> futureResponse =
-        _callNativeRestMethod(methodName, cancelToken, restOptions);
+  Future<AWSStreamedHttpResponse> _restResponseHelper({
+    required String methodName,
+    required String path,
+    required String cancelToken,
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, dynamic>? queryParameters,
+    String? apiName,
+  }) async {
+    final bodyBytes =
+        body != null ? Uint8List.fromList(await body.single) : null;
+    final restOptions = RestOptions(
+        path: path,
+        body: bodyBytes,
+        apiName: apiName,
+        queryParameters: queryParameters as Map<String, String>?,
```

Comment:
There's a reason it's <String, dynamic> and not <String, String> in the request constructors. Same is true for AWSHttpRequest and would be important to have here

Reply:
Sure, ideally this cast is not necessary. The problem is that the existing platform channel implementation uses <String, String> in `RestOptions` which is used to enforce a contract on the native side. In android library, <String, String> is used https://github.com/aws-amplify/amplify-android/blob/main/core/src/main/java/com/amplifyframework/api/rest/RestOptions.java#L36 so I don't see a great way to handle the fact that the new dart interface is more permissive than the platform channel implementation actually allows. Casting here was 1st pass solution. It could be put in a try/catch that gives the caller a better error message. Do you have any other suggestions?

Reply:
This would be better than throwing. But it's still incorrect for lists and null values.

```dart
Uri(queryParameters: queryParameters).queryParameters
```

Reply:
discussed offline and changed